### PR TITLE
[DW-1139] - Create initial version of reporting api

### PR DESCRIPTION
**Pull request by aferraromedina**
Fixes: Jira Ticket [DW-1139](https://makingsense.atlassian.net/browse/DW-1139)

### Checklist:

- [x] I have paid attention to this PR title and description
- [x] I have performed a self-review of my code
- [x] I have built it locally (or my changes does not affect the build)
- [x] I have checked all tests still run ok at Doppler.ReportingApiTest project
- [x] I have added at least one simple unit test covering the new code

### CODE REVIEW

File: Doppler.reportingApi/Controllers/DetailController.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Doppler.ReportingApi.Controllers
+{
+    [ApiController]
+    public class DetailController : ControllerBase
```
Comment (andresmoschini) [inline Doppler.reportingApi/Controllers/DetailController.cs]: What do you think about using POCO controllers?

```suggestion
    public class DetailController
```

File: Doppler.reportingApi/Controllers/SummaryController.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Doppler.ReportingApi.Controllers
+{
+    [ApiController]
+    public class SummaryController : ControllerBase
```
Comment (andresmoschini) [inline Doppler.reportingApi/Controllers/SummaryController.cs]: The same

```suggestion
    public class SummaryController
```

File: Doppler.reportingApi/Controllers/SummaryController.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        /// <summary>
+        /// Return an object summarizing the campaingns performance of an user
+        /// </summary>
+        /// <param name="accountName">User name</param>
+        [HttpGet]
+        [Route("{accountName}/summary/campaigns")]
+        [ProducesResponseType(typeof(CampaignsSummary), 200)]
+        [Produces("application/json")]
+        public async Task<IActionResult> Campaigns(string accountName, [FromQuery] BasicDatefilter dateFilter)
+        {
```
Comment (andresmoschini) [inline Doppler.reportingApi/Controllers/SummaryController.cs]: Could you include some tests for this functionality?

**Comment by andresmoschini** [comment]
You will need DopplerSecurity soon. You can add it later.

An alternative is cloning HelloMicroservice completely as it is (without changes) then change the controllers based in your requirements and also update the tests based on that.

File: Doppler.reportingApi/Doppler.reportingApi.csproj [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    <TargetFramework>net5.0</TargetFramework>
+    <DockerDefaultTargetOS>Linux</DockerDefaultTargetOS>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <NoWarn>$(NoWarn);1591</NoWarn>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="5.0.9" />
+    <PackageReference Include="Microsoft.VisualStudio.Azure.Containers.Tools.Targets" Version="1.11.1" />
+    <PackageReference Include="MongoDB.Driver" Version="2.13.1" />
```
Comment (andresmoschini) [inline Doppler.reportingApi/Doppler.reportingApi.csproj]: Are you using mongo?

**Comment by andresmoschini** [comment]
image

Probably you missed the `.doppler-ci` file.

**Comment by andresmoschini** [comment]
Maybe you could take advantage of these instructions: https://github.com/FromDoppler/hello-microservice#how-to-use-this-project

Remember that you need to also configure GitHub and DockerHub.

File: Doppler.reportingApi/Controllers/DetailController.cs (hunk 1 above)
Comment (aferraromedina) [reply Doppler.reportingApi/Controllers/DetailController.cs]: Its a good idea :)

File: Doppler.reportingApi/Controllers/SummaryController.cs (hunk 3 above)
Comment (aferraromedina) [reply Doppler.reportingApi/Controllers/SummaryController.cs]: yes :)

File: Doppler.reportingApi/Doppler.reportingApi.csproj (hunk 4 above)
Comment (aferraromedina) [reply Doppler.reportingApi/Doppler.reportingApi.csproj]: no, that was a mistake

**Comment by andresmoschini** [comment]
image

The gitlint script is configured to use the `main` branch, but you are using the `master` branch:

image

Maybe you should consider using the `main` branch, because it is the new worldwide standard.

If you still prefer to use the `master` branch, please review all other places where we are mentioning `main`.

**Pull request merged**